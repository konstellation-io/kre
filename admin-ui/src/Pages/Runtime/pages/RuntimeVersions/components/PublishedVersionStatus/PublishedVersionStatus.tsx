import Button from 'Components/Button/Button';
import Can from 'Components/Can/Can';
import ROUTE from 'Constants/routes';
import React from 'react';
import { buildRoute } from 'Utils/routes';
import cx from 'classnames';
import styles from '../../RuntimeVersions.module.scss';
import { useParams } from 'react-router';

type Props = {
  nPublishedVersions: number;
  noVersions: boolean;
};
function PublishedVersionStatus({ noVersions, nPublishedVersions }: Props) {
  const { runtimeId } = useParams();

  let title;
  if (noVersions) {
    title =
      'There are no runtime versions. Please, upload a new version to start working on this runtime.';
  } else if (nPublishedVersions >= 1) {
    title = `${nPublishedVersions} version${
      nPublishedVersions > 1 ? 's' : ''
    } published`;
  } else {
    title = 'There is no published version';
  }

  const newVersionRoute = buildRoute.runtime(ROUTE.NEW_VERSION, runtimeId);

  return (
    <div
      className={cx(styles.activeVersion, {
        [styles['active']]: nPublishedVersions
      })}
    >
      <span className={styles.versionTitle}>{title}</span>
      <Can perform="version:edit">
        <Button
          label="ADD VERSION"
          to={newVersionRoute}
          primary
          height={30}
          style={{ borderRadius: 2 }}
        />
      </Can>
    </div>
  );
}

export default PublishedVersionStatus;
