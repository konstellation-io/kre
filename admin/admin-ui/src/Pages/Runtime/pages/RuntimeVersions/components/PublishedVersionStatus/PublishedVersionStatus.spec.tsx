import React from 'react';
import { shallow } from 'enzyme';
import PublishedVersionStatus from './PublishedVersionStatus';

jest.mock('react-router', () => ({
  useParams: jest.fn(() => ({ runtimeId: 'runtimeId' }))
}));

describe('PublishedVersionStatus', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PublishedVersionStatus nPublishedVersions={5} noVersions={false} />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
