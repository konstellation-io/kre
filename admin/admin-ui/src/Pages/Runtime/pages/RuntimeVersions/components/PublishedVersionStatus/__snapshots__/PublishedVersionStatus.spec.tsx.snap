// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishedVersionStatus matches snapshot 1`] = `
<div
  className="activeVersion active"
>
  <span
    className="versionTitle"
  >
    5 versions published
  </span>
  <Can
    perform="version:edit"
  >
    <A
      height={30}
      label="ADD VERSION"
      primary={true}
      style={
        Object {
          "borderRadius": 2,
        }
      }
      to="/runtimes/runtimeId/new_version"
    />
  </Can>
</div>
`;
